adapters/targetVideo/targetvideo.go

```diff
@@ -0,0 +1,194 @@
+					}
+
+				}
+			}
+			// Remove bidder node as required
+			delete(root, "bidder")
+		}
+
+		// If we obtained a placementId, set ext.prebid.storedrequest.id = placementId
+		if placementId != "" {
+			// Build prebid.storedrequest structure, preserving existing prebid if any
+			var prebid map[string]json.RawMessage
+			if pr, ok := root["prebid"]; ok && len(pr) > 0 {
+				_ = json.Unmarshal(pr, &prebid)
+			}
+			if prebid == nil {
+				prebid = make(map[string]json.RawMessage)
+			}
+			stored := map[string]string{"id": placementId}
+			storedRaw, _ := json.Marshal(stored)
+			prebid["storedrequest"] = storedRaw
+			prebidRaw, _ := json.Marshal(prebid)
+			root["prebid"] = prebidRaw
+		}
+
+		// Marshal back the transformed ext
+		if newExt, err := json.Marshal(root); err == nil {
+			request.Imp[i].Ext = newExt
+		}
+	}
+
+	reqJSON, err := json.Marshal(request)
+	if err != nil {
+		return nil, err
+	}
+
+	headers := http.Header{}
+	headers.Add("Content-Type", "application/json;charset=utf-8")
+	headers.Add("Accept", "application/json")
+
+	//fmt.Println("TARGET VIDEO reqJson: ", string(reqJSON))
+
+	return &adapters.RequestData{
+		Method:  "POST",
+		Uri:     a.endpoint,
+		Body:    reqJSON,
+		Headers: headers,
+		ImpIDs:  openrtb_ext.GetImpIDs(request.Imp),
+	}, nil
+}
+
+func (a *adapter) MakeBids(bidReq *openrtb2.BidRequest, unused *adapters.RequestData, httpRes *adapters.ResponseData) (*adapters.BidderResponse, []error) {
+	if httpRes.StatusCode == http.StatusNoContent {
+		return nil, nil
+	}
+	if httpRes.StatusCode == http.StatusBadRequest {
+		return nil, []error{&errortypes.BadInput{Message: fmt.Sprintf("Unexpected status code: %d. Run with request.debug = 1 for more info", httpRes.StatusCode)}}
+	}
+
+	var bidResp openrtb2.BidResponse
+	if err := jsonutil.Unmarshal(httpRes.Body, &bidResp); err != nil {
+		return nil, []error{&errortypes.BadServerResponse{Message: fmt.Sprintf("error while decoding response, err: %s", err)}}
+	}
+
+	if len(bidResp.SeatBid) == 0 {
+		return nil, nil
+	}
+
+	if len(bidResp.SeatBid[0].Bid) == 0 {
+		return nil, nil
+	}
+
+	br := adapters.NewBidderResponse()
+	errs := []error{}
+
+	for _, sb := range bidResp.SeatBid {
+		for i := range sb.Bid {
+			bid := sb.Bid[i]
+			// Ensure imp exists and is video
+			mediaType := openrtb_ext.BidTypeVideo
+			for _, imp := range bidReq.Imp {
+				if imp.ID == bid.ImpID {
+					if imp.Video == nil {
+						// Not a video impression; skip
+						errs = append(errs, &errortypes.BadServerResponse{Message: fmt.Sprintf("ignoring bid id=%s for non-video imp id=%s", bid.ID, bid.ImpID)})
+						mediaType = ""
+					}
+					break
+				}
+			}
+
+			br.Bids = append(br.Bids, &adapters.TypedBid{Bid: &bid, BidType: mediaType})
+		}
+	}
+	return br, errs
+}
+
+func validateImpAndSetExt(imp *openrtb2.Imp) (int, error) {
+	if imp.Video == nil {
+		return 0, &errortypes.BadInput{Message: fmt.Sprintf("Only video impressions are supported by targetvideo. ImpID=%s", imp.ID)}
+	}
```

Comment:
You can remove this check. In your yaml file you only declared support for the video media type, in which case your adapter will only ever be called if video is not nil.